spec/features/bulkrax_import_spec.rb

```diff
@@ -281,6 +274,153 @@
+          let!(:work_2) { create :work, source_identifier: "external-id-2", doi: [doi], doi_status_when_public: "draft" }
+          let!(:work_3) { create :work, source_identifier: "external-id-3", doi: [doi], doi_status_when_public: "registered" }
+          let!(:work_4) { create :work, source_identifier: "external-id-4", doi: [doi], doi_status_when_public: "findable" }
+
+          it "does not mint a new DOI" do
+            perform_enqueued_jobs(only: [Bulkrax::ImporterJob, Hyrax::DOI::RegisterDOIJob]) do
+              Bulkrax::ImporterJob.perform_now(importer.id)
+            end
+
+            expect(work_1.reload.doi).to be_empty
+            expect(work_2.reload.doi).to eq([doi])
+            expect(work_3.reload.doi).to eq([doi])
+            expect(work_4.reload.doi).to eq([doi])
+
+            expect(work_1.doi_status_when_public).to be_nil
+            expect(work_2.doi_status_when_public).to eq("draft")
+            expect(work_3.doi_status_when_public).to eq("registered")
+            expect(work_4.doi_status_when_public).to eq("findable")
+          end
+        end
+
+        context "the import changes the DOI status forwards" do
+          let!(:work_1) { create :work, source_identifier: "external-id-1", doi: nil, doi_status_when_public: nil }
+          let!(:work_2) { create :work, source_identifier: "external-id-2", doi: nil, doi_status_when_public: nil }
+          let!(:work_3) { create :work, source_identifier: "external-id-3", doi: [doi], doi_status_when_public: "draft" }
+          let!(:work_4) { create :work, source_identifier: "external-id-4", doi: [doi], doi_status_when_public: "registered" }
+
+          it "changes DOI status and mints only for the correct items" do
+            perform_enqueued_jobs(only: [Bulkrax::ImporterJob, Hyrax::DOI::RegisterDOIJob]) do
+              Bulkrax::ImporterJob.perform_now(importer.id)
+            end
+
+            expect(work_1.reload.doi).to be_empty
+            expect(work_2.reload.doi).to eq([doi])
+            expect(work_3.reload.doi).to eq([doi])
+            expect(work_4.reload.doi).to eq([doi])
+
+            expect(work_1.doi_status_when_public).to be_nil
+            expect(work_2.doi_status_when_public).to eq("draft")
+            expect(work_3.doi_status_when_public).to eq("registered")
+            expect(work_4.doi_status_when_public).to eq("findable")
+          end
+        end
+
+        context "the import changes the DOI status backwards" do
+          let!(:work_1) { create :work, source_identifier: "external-id-1", doi: [doi], doi_status_when_public: "draft" }
+          let!(:work_2) { create :work, source_identifier: "external-id-2", doi: [doi], doi_status_when_public: "registered" }
+          let!(:work_3) { create :work, source_identifier: "external-id-3", doi: [doi], doi_status_when_public: "findable" }
+          let!(:work_4) { create :work, source_identifier: "external-id-4", doi: [doi], doi_status_when_public: "findable" }
+
```

Comment:
By taking doi backwards do yo mean to what is  in the imported csv as against the values in work-1 to work-4?

Reply:
Yes, so there are two tests here which could be rolled into one.  
(a) the import does not change the DOI status
(b) the import does not change the DOI status backwards (my mistake corrected)
Since there are 4 rows in the CSV (I could add more....) I think one test could cover both of these current tests.

Reply:
@thenapking is failing on codeclimate, should i approve, since it is not related to your code and i have not seen this failure before, hence might be something flaky.

Reply:
Sounds good :+1:

Reply:
I have improved the specs here, if you would like to re-review: [a7ce3d1](https://github.com/ubiquitypress/hyku_addons/pull/605/commits/a7ce3d10626e1ac1e6d14ccba2f1848455f3d662)